src/main/java/org/rutebanken/tiamat/importer/initial/ParallelInitialStopPlaceImporter.java

```diff
@@ -49,26 +56,82 @@ public class ParallelInitialStopPlaceImporter {
     @Value("${changelog.publish.enabled:false}")
     private boolean publishChangelog;
 
+    @Autowired
+    private StopPlaceParentCreator parentStopPlaceCreator;
+
     public List<org.rutebanken.netex.model.StopPlace> importStopPlaces(List<StopPlace> tiamatStops, AtomicInteger stopPlacesCreated) {
+        Map<String, Set<String>> childrenByParentSiteReference = new ConcurrentHashMap<>();
+        Set<StopPlace> parentStopPlaces = ConcurrentHashMap.newKeySet();
 
-        if (publishChangelog){
+        if (publishChangelog) {
             throw new IllegalStateException("Initial import not allowed with changelog publishing enabled! Set changelog.publish.enabled=false");
         }
 
-        return tiamatStops.parallelStream()
+        List<StopPlace> stops = tiamatStops.parallelStream()
                 .map(stopPlace -> {
-
-                    if(stopPlace.getTariffZones() != null) {
+                    if (stopPlace.getTariffZones() != null) {
                         stopPlace.getTariffZones().forEach(tariffZoneRef -> tariffZoneRef.setVersion(null));
                     }
-
                     return stopPlace;
                 })
                 .peek(stopPlace -> stopPlaceTopographicPlaceReferenceUpdater.updateTopographicReference(stopPlace))
-                .map(stopPlace -> stopPlaceVersionedSaverService.saveNewVersion(stopPlace))
-                .peek(stopPlace -> stopPlacesCreated.incrementAndGet())
-                .map(stopPlace -> netexMapper.mapToNetexModel(stopPlace))
-                .collect(toList());
+                .flatMap(stopPlace -> {
+                    SiteRefStructure parentSiteRef = stopPlace.getParentSiteRef();
+                    stopPlace.setParentSiteRef(null);
+
+                    if (isParent(stopPlace, parentSiteRef)) {
+                        parentStopPlaces.add(stopPlace);
+                        return empty();
+                    }
+
+                    StopPlace saved = stopPlaceVersionedSaverService.saveNewVersion(stopPlace);
+                    stopPlacesCreated.incrementAndGet();
+                    if (isChild(stopPlace, parentSiteRef)) {
+                        childrenByParentSiteReference
+                                .computeIfAbsent(parentSiteRef.getRef(), k -> newKeySet())
+                                .add(saved.getNetexId());
+                    }
+                    return of(saved);
+                }).toList();
+
+        List<StopPlace> parents = resolveParents(parentStopPlaces, childrenByParentSiteReference, stopPlacesCreated);
+
+        return concat(stops.stream(), parents.stream())
+                .map(netexMapper::mapToNetexModel)
+                .toList();
     }
 
+    private boolean isParent(StopPlace stopPlace, SiteRefStructure parentSiteRef) {
+        return stopPlace.isParentStopPlace() || (!hasQuays(stopPlace) && parentSiteRef == null);
```

Comment:
A stop with no quays AND no parent reference is classified as a parent. This seems incorrect - it might just be an incomplete or standalone stop without quays. This could lead to orphaned stops being incorrectly classified and potentially dropped.